ibm/service/globaltagging/resource_ibm_iam_access_tag.go

```diff
@@ -0,0 +1,165 @@
+// Copyright IBM Corp. 2017, 2024 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package globaltagging
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/validate"
+	"github.com/IBM/platform-services-go-sdk/globaltaggingv1"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceIBMIamAccessTag() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMIamAccessTagCreate,
+		Read:     resourceIBMIamAccessTagRead,
+		Delete:   resourceIBMIamAccessTagDelete,
+		Importer: &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.InvokeValidator("ibm_iam_access_tag", "name"),
+				Set:          flex.ResourceIBMVPCHash,
+				Description:  "Name of the access tag",
+			},
+			tagType: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Type of the tag(access)",
+			},
+		},
+	}
+}
+
+func ResourceIBMIamAccessTagValidator() *validate.ResourceValidator {
+
+	validateSchema := make([]validate.ValidateSchema, 0)
+
+	validateSchema = append(validateSchema,
+		validate.ValidateSchema{
+			Identifier:                 "name",
+			ValidateFunctionIdentifier: validate.ValidateRegexpLen,
+			Type:                       validate.TypeString,
+			Required:                   true,
+			Regexp:                     `^([A-Za-z0-9_.-]|[A-Za-z0-9_.-][A-Za-z0-9_ .-]*[A-Za-z0-9_.-]):([A-Za-z0-9_.-]|[A-Za-z0-9_.-][A-Za-z0-9_ .-]*[A-Za-z0-9_.-])$`,
+			MinValueLength:             1,
+			MaxValueLength:             128})
+
+	ibmIamAccessTagValidator := validate.ResourceValidator{ResourceName: "ibm_iam_access_tag", Schema: validateSchema}
+	return &ibmIamAccessTagValidator
+}
+
+func resourceIBMIamAccessTagCreate(d *schema.ResourceData, meta interface{}) error {
+
+	gtClient, err := meta.(conns.ClientSession).GlobalTaggingAPIv1()
+	if err != nil {
+		return fmt.Errorf("Error getting global tagging client settings: %s", err)
+	}
+
+	tagName := d.Get("name").(string)
+	add := make([]string, 0)
+	add = append(add, tagName)
+	accessTagType := "access"
+	createTagOptions := &globaltaggingv1.CreateTagOptions{
+		TagType:  &accessTagType,
+		TagNames: add,
+	}
+	results, _, err := gtClient.CreateTag(createTagOptions)
+	if err != nil {
+		return err
+	}
+	if results != nil {
+		errMap := make([]globaltaggingv1.CreateTagResultsResultsItem, 0)
+		for _, res := range results.Results {
+			if res.IsError != nil && *res.IsError {
+				errMap = append(errMap, res)
+			}
+		}
+		if len(errMap) > 0 {
+			output, err := json.MarshalIndent(errMap, "", "    ")
+			log.Printf("err is %s", err)
+			return fmt.Errorf("[ERROR] Error while creating access tag(%s) : %s", tagName, string(output))
```

Comment:
Can you upgrade the error logging mechanism as per new error framework
communicated recently Terraform slack channels
https://ibm-cloudplatform.slack.com/archives/C53NF5MB4/p1722606201652699

Reply:
The changes have been pushed. Can you re-check? Thank you